.eslintrc.js

```diff
@@ -0,0 +1,204 @@
+        'no-nested-ternary': 'error',
+        'no-new-object': 'error',
+        'no-spaced-func': 'error',
+        'no-trailing-spaces': 'error',
+        'no-whitespace-before-property': 'error',
+        'one-var': ['error', 'never'],
+        'operator-linebreak': ['error', 'after'],
+        'padded-blocks': ['error', {
+            'blocks': 'never',
+            'switches': 'never',
+            'classes': 'always'
+        }],
+        'quotes': ['error', 'single'],
+        'semi-spacing': ['error', {'before': false, 'after': true}],
+        'semi': ['error', 'always'],
+        'space-before-blocks': ['error', 'always'],
+        'space-before-function-paren': ['error', {'anonymous': 'always', 'named': 'never'}],
+        'spaced-comment': ['error', 'always', {'exceptions': ['-*'] }],
+        'space-in-parens': ['error', 'never'],
+        'wrap-regex': 'error',
+
+        // ECMAScript 2015
+        'arrow-parens': ['error', 'always'],
+        'arrow-spacing': 'error',
+        'constructor-super': 'error',
+        'generator-star-spacing': ['error', {'before': true, 'after': false}],
+        'no-class-assign': 'error',
+        'no-const-assign': 'error',
+        'no-dupe-class-members': 'error',
+        'no-duplicate-imports': 'error',
+        'no-new-symbol': 'error',
+        'no-this-before-super': 'error',
+        'no-unused-vars': ['error', {'varsIgnorePattern': '_'}],
+        'no-useless-computed-key': 'error',
+        'no-useless-constructor': 'error',
+        'no-useless-rename': 'error',
+        'no-var': 'error',
+        'object-shorthand': ['error', 'always', {'avoidQuotes': true}],
+        'prefer-arrow-callback': 'error',
+        'prefer-const': ['error', {
+            'destructuring': 'any',
+            'ignoreReadBeforeAssign': false
+        }],
+        'prefer-reflect': ['error', {'exceptions': ['delete']}],
+        'prefer-rest-params': 'error',
+        'prefer-spread': 'error',
+        'prefer-template': 'error',
+        'react-hooks/rules-of-hooks': 'error',
+        'react-hooks/exhaustive-deps': 'error',
+        'require-yield': 'error',
+        'rest-spread-spacing': ['error', 'never'],
+        'template-curly-spacing': ['error', 'never'],
+        'yield-star-spacing': ['error', {'before': false, 'after': true}],
+
+        'react/jsx-curly-newline': ['error', 'consistent'],
+        'react/jsx-curly-brace-presence': ['error', 'never'],
+        'react/jsx-equals-spacing': ['error', 'never'],
+        'react/jsx-first-prop-new-line': ['error', 'multiline'],
+        'react/jsx-indent': ['error', 4],
+        'react/jsx-indent-props': ['error', 4],
+        'react/jsx-key': 'error',
+        'react/jsx-no-target-blank': 'error',
+        'react/jsx-tag-spacing': 'error',
+        'react/jsx-uses-vars': 'error',
+        'react/jsx-uses-react': 'error',
+        'react/prop-types': 'off',
+        'react/no-children-prop': 'off'
+    }
```

Comment:
wow, that's a big list.  kudos for researching them all and listing them, I've always just went with `recommends`or someone else's definitions